optuna/integration/mlflow.py

```diff
@@ -0,0 +1,137 @@
+import optuna
+from optuna import type_checking
+
+if type_checking.TYPE_CHECKING:
+    from typing import Dict  # NOQA
+
+try:
+    import mlflow
+
+    _available = True
+except ImportError as e:
+    _import_error = e
+    _available = False
+    mlflow = object
+
+
+def _check_mlflow_availability():
+    # type: () -> None
+
+    if not _available:
+        raise ImportError(
+            "Mlflow is not available. Please install Mlflow to use this "
+            "feature. It can be installed by executing `$ pip install "
+            "mlflow`. For further information, please refer to the installation guide "
+            "of Mlflow. (The actual import error is as follows: " + str(_import_error) + ")"
+        )
+
+
+class MlflowCallback(object):
```

Comment:
@PhilipMay @g-votte 
Should `Ml` be uppercase?

```suggestion
class MLflowCallback(object):
```

Reply:
I fixed that. Thanks.